lib/components/src/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,207 @@
+  const accordionMapIndex = useRef(0);
+
+  const addToMap: AddToMapFn = useCallback(
+    (id) => {
+      const index = accordionMapIndex.current + 1;
+      accordionMap.current = { ...accordionMap.current, [id]: { id, index } };
+      accordionMapIndex.current = index;
+
+      if (typeof open === 'number') {
+        openStateMap.current = { ...openStateMap.current, [id]: open === index };
+      } else if (Array.isArray(open)) {
+        const isExpanded = open.includes(index);
+        openStateMap.current = { ...openStateMap.current, [id]: isExpanded };
+      } else {
+        openStateMap.current = { ...openStateMap.current, [id]: false };
+      }
+    },
+    [accordionMap, openStateMap, open]
+  );
+
+  const onItemClose = useCallback(
+    (id: string) => {
+      setOpenState({ ...openState, [id]: false });
+
+      if (onClose) {
+        onClose({ id, index: accordionMap.current[id].index });
+      }
+    },
+    [openState, setOpenState, onClose, accordionMap]
+  );
+
+  const onItemOpen = useCallback(
+    (id: string) => {
+      const newOpen = { ...openState };
+      let oldOpen: StateChange | undefined;
+
+      if (allowMultipleOpen) {
+        newOpen[id] = true;
+      } else {
+        Object.keys(openState).forEach((key) => {
+          if (openState[key]) {
+            oldOpen = { id: key, index: accordionMap.current[key].index };
+          }
+
+          newOpen[key] = id === key;
+        });
+      }
+
+      if (oldOpen && onClose) {
+        onClose(oldOpen);
+      }
+
+      setOpenState({ ...openState, ...newOpen });
+
+      if (onOpen) {
+        onOpen({ id, index: accordionMap.current[id].index });
+      }
+    },
+    [openState, setOpenState, onOpen, accordionMap]
+  );
+
+  useEffect(() => {
+    if (Object.keys(accordionMap.current).length === Children.count(children)) {
+      setOpenState({ ...openState, ...openStateMap.current });
+    }
+  }, [accordionMap, openStateMap]);
+
+  useEffect(() => {
+    let newOpen: number[] = [];
+
+    if (typeof open === 'number') {
+      newOpen.push(open);
+    } else if (Array.isArray(open)) {
+      newOpen = open;
+    }
+
+    let newOpenState = { ...openState };
+
+    Object.keys(accordionMap.current).forEach((key) => {
+      const item = accordionMap.current[key];
+
+      newOpenState = { ...newOpenState, [item.id]: newOpen.includes(item.index) };
+    });
+
+    setOpenState({ ...openState, ...newOpenState });
+  }, [open]);
+
+  return (
+    <AccordionContext.Provider
+      value={{
+        addToMap,
+        onClose: onItemClose,
+        onOpen: onItemOpen,
+        bordered,
+        openState,
+        indentBody,
+        lined,
+        narrow,
+        preventToggle,
+      }}
+    >
+      <Wrapper data-sb-accordion="" bordered={bordered} rounded={rounded} {...rest}>
+        {children}
+      </Wrapper>
+    </AccordionContext.Provider>
+  );
+};
+
+interface WrapperProps {
+  bordered: boolean;
+  rounded: boolean;
+}
```

Comment:
try not to re-declare the same types, I think we should `Pick<>` from the `AccordionProps` interface

Reply:
They are not the same - they are optional in the props for the component and not in the styled component.

It for me would also make it harder to move this in the future - and would make a connection where no connection is required - I very much prefer to keep that stuff separated and agnostic